examples/hybrid_plugin_system.ipynb

```diff
@@ -0,0 +1,391 @@
+{
+ "cells": [
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Hybrid Plugin System\n",
+    "\n",
+    "This notebook illustrates usage of the Hybrid Plugin System. We combine a Python manager plugin and a C++ manager plugin, and dispatch to the appropriate plugin based on its capabilities and priority. \n",
+    "\n",
+    "The primary use-case for this feature is to allow performance critical functionality to be written in performant C++, whilst less performance critical functionality can be written in more flexible Python.\n",
+    "\n",
+    "The hybrid plugin system is not limited to this use-case, however. You are free to combine any number of plugins in any supported language (only C++ and Python are available in the core library at time of writing). The hybrid plugin system also provides a more convenient abstraction for working with multiple plugin systems in general.\n",
+    "\n",
+    "## How it works\n",
+    "\n",
+    "OpenAssetIO manager plugins must advertise a unique identifier. An OpenAssetIO plugin system (e.g. C++ or Python) maps unique identifiers to a plugin instance. A consequence of this is if multiple plugins advertise the same unique identifier, only one of those plugins can be chosen _by that plugin system_. However, if multiple plugin systems are in use,\n",
+    "then each plugin system has their own mapping of unique identifier to plugin instance. This means there _can_ be multiple plugins with the same identifier, as long as they are discovered by different plugin systems.\n",
+    "\n",
+    "As a motivating example, a manager plugin developed in Python, loaded by the Python plugin system, could advertise the same identifier as a manager plugin developed in C++, loaded by the C++ plugin system.\n",
+    "\n",
+    "This is the essence of how the hybrid plugin system discovers plugins. If two (or more) plugins from two (or more) different plugin systems advertise the same unique identifier, then we assume they are related and can be composed.\n",
+    "\n",
+    "The hybrid plugin system therefore wraps a list of child plugin systems, such that they present as a single plugin system to the host.\n",
+    "\n",
+    "Given that the hybrid plugin system has discovered two (or more) composable plugins, we then need a mechanism to dispatch API calls to the appropriate plugin. I.e. we need a way to choose which plugin is the best to use for a particular API call. This is where the `hasCapability` API method comes in.\n",
+    "\n",
+    "OpenAssetIO API methods are grouped under \"capabilities\", e.g. `\"resolution\"`, `\"publishing\"`, `\"relationshipQueries\"`, etc (these are stringified representations of the `Capability` enumeration). A manager plugin advertises which capabilities it supports by overriding the `hasCapability` method of the base `ManagerInterface` class.\n",
```

Comment:
Q: these strinigifications aren't used in the API though, right?

Wondering why stringification is called out at all here.

Reply:
The stringification turns up in error messages and is used in documentation here and elsewhere. I thought it was worth calling out that the camelCased names are distinct from the enum you'd use in code.